contracts/adapters/UniswapAdapter.sol

```diff
@@ -0,0 +1,233 @@
+        if (keccak256(abi.encodePacked(exchangeName)) == uniswapHash){
+                (uint112 reserve_amt,,) = IUniswapV2Pair(wbtcDiggUniswap).getReserves();
+                uint256 trade_amt = getSwapAmt(reserve_amt,wbtcAmt);
+                address[] memory path = new address[](2);
+                path[0] = wbtcAddress;
+                path[1] = diggAddress;
+                uint256 diggOut = uniswapRouter.getAmountsOut(trade_amt, path)[1];
+                return diggOut;
+        }
+        else if (keccak256(abi.encodePacked(exchangeName)) == sushiswapHash){
+                (uint112 reserve_amt,,) = IUniswapV2Pair(wbtcDiggSushiswap).getReserves();
+                uint256 trade_amt = getSwapAmt(reserve_amt,wbtcAmt);
+                address[] memory path = new address[](2);
+                path[0] = wbtcAddress;
+                path[1] = diggAddress;
+                uint256 diggOut = uniswapRouter.getAmountsOut(trade_amt, path)[1];
+                return diggOut;
+        }
+    }
+
+    function convertEthToToken( uint256 inputAmount, address addr,uint256 amountOutMin, Exchange exchange) internal {
+        IUniswapV2Router02 router = exchange == Exchange.Uniswap ? uniswapRouter : sushiswapRouter;
+        _convertEthToToken(inputAmount, addr, amountOutMin, router);
+    }
+
+    function convertTokenToToken(address addr1, address addr2, uint256 amount, uint256 amountOutMin, Exchange exchange) internal {
+        IUniswapV2Router02 router = exchange == Exchange.Uniswap ? uniswapRouter : sushiswapRouter;
+        _convertTokenToToken(addr1, addr2, amount, amountOutMin, router);
+    }
+
+    function addLiquidity(address token1,address token2,uint256 amount1,uint256 amount2,Exchange exchange) internal {
+        IUniswapV2Router02 router = exchange == Exchange.Uniswap ? uniswapRouter : sushiswapRouter;
+        _addLiquidity(token1, token2, amount1, amount2, router);
+    }
+
+    function _convertEthToToken( uint256 inputAmount, address addr, uint256 amountOutMin,IUniswapV2Router02 router) internal {
+        uint deadline = block.timestamp + deadlineBuffer;
+        address[] memory path = new address[](2);
+        path[0] = wethAddress;
+        path[1] = addr;
+        router.swapExactETHForTokensSupportingFeeOnTransferTokens{value: inputAmount }(amountOutMin, path, address(this), deadline);
+    }
+
+
+    function _convertTokenToToken(address addr1, address addr2, uint256 amount, uint256 amountOutMin, IUniswapV2Router02 router) internal {
+        uint deadline = block.timestamp + deadlineBuffer;
+        IERC20 tokenIn = IERC20(addr1);
+        address[] memory path = new address[](2);
+        path[0] = addr1;
+        path[1] = addr2;
+        if (tokenIn.allowance(address(this), address(router)) == 0){
+                SafeERC20.safeApprove(tokenIn, address(router), type(uint256).max);
+        }
+
+        router.swapExactTokensForTokens(amount,amountOutMin, path, address(this), deadline);

```

Comment:
We should return the amount that was swapped to the account, then with that value, we can avoid querying for the user's balance.

```
uint[] memory amountsOut = router.swapExactTokensForTokens(amount,amountOutMin, path, address(this), deadline);
return amountsOut[1];
```